src/middleware/checkAuth.js

```diff
@@ -0,0 +1,48 @@
+const jwt = require('jsonwebtoken');
+
+const generateToken = (response, user) => {
+  // eslint-disable-next-line camelcase
+  const { user_id, userName, email, role } = user;
+  const payload = {
```

Comment:
payload at the beginning will only  have access to userName, email and id role will be assgined after the authentication is done